### ISSUE TITLE
Doesn't work for me

### ISSUE THREAD

**Issue by hermanmonillaluna** [jkuri/ng2-datepicker#78]
![error](https://cloud.githubusercontent.com/assets/22231324/19380521/2b652e7e-922a-11e6-8d49-8c8e85699767.png)

Steps that I did

1) npm install ng2-datepicker --save

Since I used ionicons in my app. I didn't add it again but I also tried it to put in my index.html.

2) Import module

`import { DatePicker } from 'ng2-datepicker/ng2-datepicker'`

3) Used it on my component. I used reactive forms.

`<datepicker formControlName="dateFrom" [expanded]="true"></datepicker>`

I didn't change my systemjs.config.js

Possible cause of error:
Running to all components. (I'm not sure of it. I'm not going to use it for all the components) 

Any help would be appreciated. Thanks

**Comment by ghetolay** [comment]
Again duplicate of #66 and #50.

**Comment by jkuri** [comment]
Is that something that `moduleId: module.id` and `templateUrl: './ng2-datepicker.component.html'` to get relative paths should solve?

**Comment by ghetolay** [comment]
Yes it's supposed to. @shamgang post a comment about it [here](https://github.com/jkuri/ng2-datepicker/issues/66#issuecomment-253282052).

It's seems the way you write the url may make it work with some module bundler but there is no universal solution except inline. We've discuss about that on #66.

**Comment by hermanmonillaluna** [comment]
I solved it by copying the missing .html and .css to the main folder. But I think it is not good for application. Thanks tho @jkuri. :100: % Nice woooooooork

**Comment by hermanmonillaluna** [comment]
Nevermind. I got a solution.

For those who's having trouble with the same problem

Open node_modules/ng2-datepicker/src/components/ng2-datepicker.js

Vhange line 167 and 168 to:

`templateUrl: './node_modules/ng2-datepicker/src/components/ng2-datepicker.component.html',`
`styleUrls: ['./node_modules/ng2-datepicker/src/components/ng2-datepicker.css'],`

I am not sure if this is proper way to do it. But it worked with me. Try it. :dagger: :dagger:

**Comment by hermanmonillaluna** [comment]
![datepicker](https://cloud.githubusercontent.com/assets/22231324/19407611/c6ffec0c-92d9-11e6-88b9-e7a7a2e7e164.png)

Is it possible to share SASS code here?

What I did is re-write css to SASS and change some minor changes to fit on my project. Because, it's pretty hard to edit the css file. I think it is much better to be customizable. Thanks!

**Comment by ghetolay** [comment]
I think the best way to use custom css **for now** would be something like that : 

```ts
import { DatePicker } from 'ng2-datepicker';

let DatePickerComponent = DatePicker[0];

@Component({
    selector: 'my-datepicker',
    templateUrl: './node_modules/ng2-datepicker/src/components/ng2-datepicker.component.html',
    styleUrls: [ './mydatepicker.css' ]
})
export class MyDatepicker extends DatePickerComponent { }
```

You may need to adapt template/style to your build system but that's the spirit.

We're working on a better way to customize DatePicker.

**Comment by hermanmonillaluna** [comment]
Are you going to support it with SASS?

**Comment by ghetolay** [comment]
What you use to build your custom css is up to you and your build system. The same way you would do for any Angular 2 component.

**Comment by hermanmonillaluna** [comment]
Well, you're right. But what I mean is, what if I just want to change a small piece of the design? The design you provide is great. It's cathcy in eyes. That's what people love and clients too. I need to look for the element reference then update the design. For new programmer, that will be a little bit difficult since it has a lot of style in it. That's only a suggestion tho. Thanks for your work :) really helped a lot including me, myself, and I.

**Comment by ghetolay** [comment]
`styles` property of Component is an array, so I guess you can just add a .css file to ours : 
```
@Component({
    ....
     styles: ['./node_modules/ng2-datepicker/src/components/ng2-datepicker.css.', './mydatepicker.css']
})
```

Would that work for you ?

**Comment by ensaier** [comment]
The solution with NG_VALUE_ACCESSOR doesn't work for me. I'm getting 'Type 'any[]' is not a constructor function type.' I've tried to pass elementRef to a constructor, but it didn't help.

**Comment by ghetolay** [comment]
@ensaier it should work can we see your code ?

**Comment by ensaier** [comment]
@ghetolay thanks for your response, but I've already solved a problem in the other way, and lost those lines of code. We had to implement a bunch of custom functionality so we've decided to proceed with a custom solution.
